### DISCUSSION TITLE
Best alternative to "grid-template-areas" in tailwind

### DISCUSSION THREAD

**Discussion by udit99** [tailwindlabs/tailwindcss#2784] [Help] [answered]
Hey there
Im trying to achieve [this layout](https://jsfiddle.net/udit99/y0bxwdzt/56/) using tailwind. Specifically I'm looking to model this layout:
```
  .wrapper {
    text-align: initial;
    grid-template-areas:
     "hero-image title"
     "hero-image subheading"
     "hero-image details-list"
     "hero-image cta";
  }
```
I haven't been able to figure out how to use `col-span-x` to get the same effect (Div spanning a column that cuts through multiple rows)

**Comment by horuskol** [comment, marked as answer]
Hi,

I put together a plugin that creates utility classes when you configure `grid-template-areas`:

https://www.npmjs.com/package/@savvywombat/tailwindcss-grid-areas

**Comment by donhector** [comment]
Maybe this will help? https://stackoverflow.com/questions/67242334/tailwind-css-how-to-make-a-grid-with-two-columns-where-the-1st-column-has-20

**Reply by mendrik** [reply]
I use styled components for defining the areas and `className="[grid-area:header]"`  etc for the items

**Comment by jd-solanki** [comment]
Now you can do this: `[grid-template-areas:'header_header_header''main_main_aside''footer_footer_footer']`

![image](https://github.com/tailwindlabs/tailwindcss/assets/47495003/22eb815d-fb65-4cf9-8e5e-58fe404bce40)

**Reply by futile** [reply]
In tailwindcss itself? :thinking: Are there docs on the website already? :open_mouth:

**Reply by wongjn** [reply]
Yes, this is an example of an arbitrary property class. [See documentation on the website](https://tailwindcss.com/docs/adding-custom-styles#arbitrary-properties).

**Reply by futile** [reply]
Ah, thank you, perfect! :)

**Reply by jack3898** [reply]
Does anyone know why this doesn't seem to work with other values? When I try `leftpanel` as an area name, tailwind falls over and no longer generates the style?

**Reply by wongjn** [reply]
Seems to work in [this Tailwind Play](https://play.tailwindcss.com/2P7GAasW25) (open the *Generated CSS* panel). Does your class work in Tailwind Play?

**Reply by jack3898** [reply]
Ah, I classic blunder. I was doing this:
![image](https://github.com/tailwindlabs/tailwindcss/assets/28375223/a2562c15-670c-4c9c-91b7-ce7e9687b0e0)
Of course it's string interpolation. Tailwind 101. 🤦‍♂️
Mixed with some class names persisting from when I didn't do that, explains why it seemed like some grid area names were working while others were not.
My bad! Thanks for the super quick reply @wongjn! Appreciated! 🙌

**Reply by airtonix** [reply]
@jack3898 if you really wanted to do that you could use css vars there.
